src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceRecaptcha.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <v-input :model-value="modelValue" :rules="props.rules" hide-details="auto">
+    <div
+      :id="props.recaptchaElementId"
+      class="g-recaptcha"
+      :data-sitekey="siteKey"
+      data-callback="recaptchaSuccessCallback"
+      data-expired-callback="recaptchaExpiredCallback"
+    ></div>
+  </v-input>
+</template>
+<script lang="ts">
+import { useScriptTag } from "@vueuse/core";
+import { defineComponent } from "vue";
+
+import { getRecaptchaSiteKey } from "@/api/configuration";
+import type { EceRecaptchaProps } from "@/types/input";
+
+export default defineComponent({
+  name: "EceRecaptcha",
+  props: {
+    modelValue: {
+      type: String,
+      required: true,
+    },
+    props: {
+      type: Object as () => EceRecaptchaProps,
+      required: true,
+    },
+  },
+  emits: {
+    "update:model-value": (_recaptchaToken: string) => true,
+  },
+  async setup() {
+    const siteKey = await getRecaptchaSiteKey();
+    useScriptTag("https://www.recaptcha.net/recaptcha/api.js?onload=recaptchaOnloadCallback&render=explicit", () => {}, { async: true, defer: true });
```

Comment:
Cool, never seen this hook

Reply:
Yeah it's pretty interesting, loads scripts only when you use the component.